types.d.ts

```diff
@@ -74,142 +138,305 @@ export interface UserFromGetMe extends User {
+    pinned_message?: Message;
+  }
+  /** Internal type holding properties that those group and supergroup chats returned from `getChat` share. */
+  interface MultiUserGetChat extends NonPrivateGetChat {
+    /** Default chat member permissions, for groups and supergroups. Returned only in getChat. */
+    permissions?: ChatPermissions;
+    /** True, if the bot can change the group sticker set. Returned only in getChat. */
+    can_set_sticker_set?: Boolean;
+  }
+
+  // ==> GET CHATS
+  /** Internal type representing private chats returned from `getChat`. */
+  export interface PrivateGetChat extends PrivateChat, GetChat {}
+  /** Internal type representing group chats returned from `getChat`. */
+  export interface GroupGetChat extends GroupChat, MultiUserGetChat {}
+  /** Internal type representing supergroup chats returned from `getChat`. */
+  export interface SupergroupGetChat extends SupergroupChat, MultiUserGetChat {
+    /** For supergroups, the minimum allowed delay between consecutive messages sent by each unprivileged user. Returned only in getChat. */
+    slow_mode_delay?: Integer;
+    /** For supergroups, name of group sticker set. Returned only in getChat. */
+    sticker_set_name?: String;
+  }
+  /** Internal type representing channel chats returned from `getChat`. */
+  export interface ChannelGetChat extends ChannelChat, NonPrivateGetChat {}
 }
+
+/** This object represents a chat. */
+export type Chat =
+  | Chat.PrivateChat
+  | Chat.GroupChat
+  | Chat.SupergroupChat
+  | Chat.ChannelChat;
+
 /** This object represents a Telegram user or bot that was returned by `getChat`. */
-export interface ChatFromGetChat extends Chat {
-  /** Chat photo. Returned only in getChat. */
-  photo?: ChatPhoto;
-  /** Description, for groups, supergroups and channel chats. Returned only in getChat. */
-  description?: String;
-  /** Chat invite link, for groups, supergroups and channel chats. Each administrator in a chat generates their own invite links, so the bot must first generate the link using exportChatInviteLink. Returned only in getChat. */
-  invite_link?: String;
-  /** Pinned message, for groups, supergroups and channels. Returned only in getChat. */
-  pinned_message?: Message;
-  /** Default chat member permissions, for groups and supergroups. Returned only in getChat. */
-  permissions?: ChatPermissions;
-  /** For supergroups, the minimum allowed delay between consecutive messages sent by each unpriviledged user. Returned only in getChat. */
-  slow_mode_delay?: Integer;
-  /** For supergroups, name of group sticker set. Returned only in getChat. */
-  sticker_set_name?: String;
-  /** True, if the bot can change the group sticker set. Returned only in getChat. */
-  can_set_sticker_set?: Boolean;
+export type ChatFromGetChat =
+  | Chat.PrivateGetChat
+  | Chat.GroupGetChat
+  | Chat.SupergroupGetChat
+  | Chat.ChannelGetChat;
+
+declare namespace Message {
```

Comment:
These types will be very useful for https://github.com/telegraf/telegraf/pull/1116
```suggestion
export namespace Message {
```

Reply:
We could generally just export the namespaces. There's not really any “internal logic” in this lib that needs to be hidden. If someone wants to dig into how that types are built and leverage that structure, I don't see why we should prevent that by not exporting something.